pinot-core/src/main/java/org/apache/pinot/core/common/datablock/DataBlockFactory.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.common.datablock;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.pinot.core.common.datatable.DataTableBuilder;
+import org.apache.pinot.core.common.datatable.DataTableImplV4;
+
+
+public class DataBlockFactory {
```

Comment:
Do we need this factory? We should use `DataTableFactory`

Reply:
this is used in `testDataBlockBuilderV4()` and it returns `BaseDataBlock` instead of `DataTable` so I can work with and test `DataBlock` directly.

Reply:
If it is not used in production code, I'd suggest not adding it for now. You may use `DataTableFactory` and cast the `DataTable` to a `BaseDataBlock`